spec/bambora/rest/batch_payment_file_upload_client_spec.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+module Bambora
+  module Rest
+    describe BatchPaymentFileUploadClient do
+      subject(:client) do
+        described_class.new(base_url: base_url, merchant_id: merchant_id, sub_merchant_id: sub_merchant_id)
+      end
+
+      let(:merchant_id) { 1 }
+      let(:sub_merchant_id) { 2 }
+      let(:base_url) { 'https://sandbox-api.na.bambora.com' }
+      let(:api_key) { 'fakekey' }
+      let(:csv_content) do
+        <<~CSV
+          E,C,001,99001,09400313371,10000,1000070001,ACME Corp
+          E,C,002,99002,09400313372,20000,1000070002,John Doe
+          E,C,003,99003,09400313373,30000,1000070003,Jane Doe
+        CSV
+      end
+      let(:options) { { process_now: 1 } }
+      let(:response_headers) { { 'Content-Type' => 'application/json' } }
+      let(:response_body) do
+        {
+          code: 1,
+          message: 'File successfully received',
+          batch_id: '100',
+          process_date: Date.today.to_s,
+          process_time_zone: 'GMT-08:00',
+          batch_mode: 'live',
+        }
+      end
+
+      let(:request_body) do
+        <<~BODY
+          --multiparty-boundary-50428029
+          Content-Disposition: form-data; name="criteria"
+          Content-Type: application/json
+
+          {"process_now":1}
+          --multiparty-boundary-50428029
+          Content-Disposition: form-data; name="data"; filename="merchant_2.txt"
+          Content-Type: text/plain
+          Content-Transfer-Encoding: binary
+
+          E,C,001,99001,09400313371,10000,1000070001,ACME Corp
+          E,C,002,99002,09400313372,20000,1000070002,John Doe
+          E,C,003,99003,09400313373,30000,1000070003,Jane Doe
+
+          --multiparty-boundary-50428029--
+        BODY
```

Comment:
Thank you for putting this in the code. As far as I know, I have only seen this in Bambora's example request in the Postman requests. Great to have this here too.